### DISCUSSION TITLE
Drawer close animation

### DISCUSSION THREAD

**Discussion by sudoDeznit** [chakra-ui/chakra-ui#4327] [Q&A] [answered]
I created a sidebar that converts to drawer on small screens.
The open animation works perfectly, but the close animation is not working.

The way I passed down onClose from useDisclosure is it the right way of doing it ??

CodeSandbox
https://codesandbox.io/s/quizzical-fast-l8r6r?file=/src/components/Sidebar.tsx:5025-5038

**Comment by akrifari** [comment, marked as answer]
Adding `DrawerOverlay` component seems address the problem.

CodeSandbox:
https://codesandbox.io/s/dazzling-napier-o46ik?file=/src/components/Sidebar.tsx:5367-5418

**Reply by sudoDeznit** [reply]
it does work after adding DrawerOverlay. Thanks
Do you know if its possible to control the transition speed of drawer?

**Reply by akrifari** [reply]
Apologies for the delay, I am not sure if we can play around with `Drawer` transition speed at the moment (CMIIW) but if you don't mind replacing `Drawer` with `Slide` then you can use `Slide`'s  `style` prop to achieve that. Unfortunately, there are drawbacks that you have to consider (focus trap, accessibility, etc).

Here's the demo:
https://codesandbox.io/s/affectionate-khayyam-gh5mm?file=/src/components/Sidebar.tsx:5268-5447

**Reply by sudoDeznit** [reply]
Thank You. I think it's better to stick with drawer then.
